code/projects/app/src/app/shared/image-display/image-display.component.ts

```diff
@@ -1,20 +1,69 @@
-import { Component, effect, ElementRef, input, viewChild } from "@angular/core";
+import {
+  animate,
+  state,
+  style,
+  transition,
+  trigger,
+} from "@angular/animations";
+import {
+  Component,
+  effect,
+  ElementRef,
+  input,
+  signal,
+  viewChild,
+} from "@angular/core";
 import { drawOn } from "../../util/image-data";
+import { ImageComparisonService } from "../image-comparison/image-comparison.service";
+
+type Flash = "red" | "green" | "default";
 
 @Component({
   selector: "app-image-display",
   templateUrl: "./image-display.component.html",
   styleUrl: "./image-display.component.css",
+  animations: [
+    trigger("flash", [
+      state("red", style({ backgroundColor: "var(--app-red)" })),
+      state("green", style({ backgroundColor: "var(--app-green)" })),
+      state("*", style({ backgroundColor: "var(--app-navy)" })),
+      transition("red => *", animate("300ms")),
+      transition("green => *", animate("300ms")),
+      transition("* => red", animate("0ms")),
+      transition("* => green", animate("0ms")),
+    ]),
+  ],
 })
 export class ImageDisplayComponent {
   public readonly image = input<ImageData>();
+  public showButtons = input<boolean>(true);
+  protected showControls = signal(false);
 
   private readonly canvas =
     viewChild.required<ElementRef<HTMLCanvasElement>>("canvas");
 
-  public constructor() {
+  protected state = { left: "default" as Flash, right: "default" as Flash };
+
+  constructor(private readonly imageLocalStorage: ImageComparisonService) {
```

Comment:
**nit:** How about `inject` instead of constructor injection? Both are fine with me, but I've seen [more and more people use `inject` recently](https://www.reddit.com/r/Angular2/comments/12kk84j/why_i_should_use_angular_14s_new_inject_instead/).